tests/core/editable/insertion.js

```diff
@@ -193,11 +193,21 @@
 			expected: '<span>&nbsp;</span><div>div</div><span>bar</span>',
 			ignore: CKEDITOR.env.ie && !CKEDITOR.env.edge // (#3061)
 		}, {
-			name: 'when empty element is inserted ad the end of span',
+			name: 'when empty element is inserted at the end of span',
 			initial: '<span>foo{bar}</span>',
 			data: '<div></div>',
 			expected: '<span>foo</span><div></div>',
 			ignore: CKEDITOR.env.ie && CKEDITOR.env.version < 9 // IE8 fills empty element with `&nbsp;`, so we can skip this test.
+		}, {
+			name: 'when selection covers nested span',
+			initial: '<p><span>{foobar}</span></p>',
+			data: '<div>div</div>',
+			expected: '<div>div</div>'
+		}, {
+			name: 'when selection covers span with empty spans on boundaries',
+			initial: '<span></span><span>{foobar}</span><span></span>',
```

Comment:
This test passes for me, but I'm not sure about it. Empty spans have no purpose.

- They are removed on `setData`, so going to source and back to wysiwyg makes them disappear.
- You can't place caret in empty span.
- https://github.com/ckeditor/ckeditor-dev/blob/cb1dd2945a20ba481b5f57560fb0bb96ae2d739c/core/dom/range.js#L724-L726

Reply:
This test also passes for me :D I have added it because empty `spans` are removed by the mechanism introduced in this PRa. So its purpose is to make sure that empty spans not added during splitting are not removed. But if you think it's redundant feel free to remove it.

The failing test (mentioned in https://github.com/ckeditor/ckeditor-dev/pull/2987#pullrequestreview-240620712) is the one above - `when selection covers nested span`.

Reply:
It seems to work on all browsers, so I'm keeping it.